### PR TITLE
[Core/Spells]Warrior, fixed bladestorm aura remove due to weapon swap

### PR DESCRIPTION
**Pull request by Tiuz90**
https://github.com/TrinityCore/TrinityCore/issues/2909
It works as intended, but it could be improved (es. check if the removed item meets the requirements of the new equipped item ---> skip aura removing)

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -12150,7 +12150,8 @@ void Player::RemoveItem(uint8 bag, uint8 slot, bool update)
                 // remove item dependent auras and casts (only weapon and armor slots)
                 if (slot < EQUIPMENT_SLOT_END)
                 {
-                    RemoveItemDependentAurasAndCasts(pItem);
+                    if(update)
```
Comment (P-Kito) [inline src/server/game/Entities/Player/Player.cpp]: if (update)     a little coding style mistake :)

Comment (Goatform) [inline src/server/game/Entities/Player/Player.cpp]: `if (update)`

**Comment by Aokromes** [comment]
Any reason to don't merge this?

**Comment by Kittnz** [comment]
Can we can some feedback for this one please?

**Comment by mik1893** [comment]
i had it live for some time and it seems to work fine.

**Pull request merged**